spring-boot-project/spring-boot-actuator-autoconfigure/src/test/java/org/springframework/boot/actuate/autoconfigure/metrics/jdbc/DataSourcePoolMetricsAutoConfigurationTests.java

```diff
@@ -185,6 +193,27 @@ private static HikariDataSource createHikariDataSource(String poolName) {
 		return hikariDataSource;
 	}
 
+	@Configuration
```

Comment:
I don't understand why initializing the pool here is necessary. Can you please clarify why this is done as part of this PR?